### PR TITLE
Added generic DispatchLevelResult to the MessageDispatchResult

### PR DESCRIPTION
**Pull request by bkontur**

### CODE REVIEW

File: bin/runtime-common/src/messages.rs [hunk 1]
```diff
@@ -449,6 +449,7 @@ pub mod target {
 		WeightCredit: Get<Weight>,
 	{
 		type DispatchPayload = FromBridgedChainMessagePayload<CallOf<ThisChain<B>>>;
+		type DispatchLevelResult = ();
```
Comment (bkontur) [inline bin/runtime-common/src/messages.rs:452]: @svyatonik 
do you want to change this to `type DispatchLevelResult = Result<Outcome, Error>` (I didnt try if compiles) or `()` is ok as default for `FromBridgedChainMessageDispatch`?

Comment (svyatonik) [reply bin/runtime-common/src/messages.rs:452]: IMHO `()` is enough for our testnets - dispatch result is not important here, only delivery

File: primitives/messages/src/lib.rs [hunk 2]
```diff
@@ -237,12 +237,12 @@ impl<Result> ReceivedMessages<Result> {
 
 /// Result of single message receival.
 #[derive(RuntimeDebug, Encode, Decode, PartialEq, Eq, Clone, TypeInfo)]
-pub enum ReceivalResult {
+pub enum ReceivalResult<DispatchLevelResult: Clone + Decode + sp_std::fmt::Debug + Eq> {
```
Comment (svyatonik) [inline primitives/messages/src/lib.rs:240]: Can this be avoided by replacing with `#[derive(RuntimeDebugNoBound, Encode, Decode, PartialEqNoBound, EqNoBound, CloneNoBound, TypeInfo]`?

File: primitives/runtime/src/messages.rs [hunk 3]
```diff
@@ -22,7 +22,7 @@ use scale_info::TypeInfo;
 
 /// Message dispatch result.
 #[derive(Encode, Decode, RuntimeDebug, Clone, PartialEq, Eq, TypeInfo)]
-pub struct MessageDispatchResult {
+pub struct MessageDispatchResult<DispatchLevelResult: Clone + Decode + sp_std::fmt::Debug + Eq> {
```
Comment (svyatonik) [inline primitives/runtime/src/messages.rs:25]: Can you also try `NoBound` versions here? :) I've found them recently && iiuc it should help with removing these constraints (maybe I'm wrong though)

**Review by svyatonik — APPROVED** [review]
LGTM

File: primitives/messages/src/lib.rs (hunk 2 above)
Comment (bkontur) [reply primitives/messages/src/lib.rs]: hmm, I just removed those `Clone + Decode + sp_std::fmt::Debug + Eq` from ReceivalResult and MessageDispatchResult and they are not needed here.
I have no idea why I added them there :)

File: primitives/runtime/src/messages.rs (hunk 3 above)
Comment (bkontur) [reply primitives/runtime/src/messages.rs]: when I tried to change `RuntimeDebug` to `RuntimeDebugNoBound` I get this error:
image

**Pull request merged**